crates/ra_ide/src/hover.rs

```diff
@@ -48,10 +77,20 @@ impl HoverResult {
         &self.results
     }
 
+    pub fn actions(&self) -> &[HoverAction] {
+        &self.actions
+    }
+
+    pub fn push_action(&mut self, action: HoverAction) {
+        self.actions.push(action);
+    }
+
     /// Returns the results converted into markup
     /// for displaying in a UI
+    ///
+    /// Does not process actions!
     pub fn to_markup(&self) -> String {
-        self.results.join("\n\n---\n")
+        self.results.join("\n\n___\n")
```

Comment:
Is this correct? We just went back and forth on this a few PRs ago.

Reply:
I believe yes, it is. As I can see '---' was replaced by '___', but not in this place.

`---` cause problems if hover content is just a single line.
```
some test
---
```

Becomes

some test
---

while 
```
some test
___
```

some test
___

Reply:
SGTM

Reply:
Btw, hover actions are rendered on the client and this method is not used for them at all.